iron-icon.html

```diff
@@ -160,9 +164,15 @@
             this._iconset = this._meta.byKey(this._iconsetName);
             if (this._iconset) {
               this._iconset.applyIcon(this, this._iconName, this.theme);
+              this.unlisten(window, 'iron-iconset-added', '_updateIcon');
+              clearTimeout(this._warningTimeout);
             } else {
-              this._warn(this._logf('_updateIcon', 'could not find iconset `'
-                + this._iconsetName + '`, did you import the iconset?'));
+              this.listen(window, 'iron-iconset-added', '_updateIcon');
+
+              this._warningTimeout = setTimeout(function() {
+                this._warn(this._logf('_updateIcon', 'could not find iconset `'
+                    + this._iconsetName + '`, did you import the iconset?'));
+              }.bind(this), 5000);
```

Comment:
Since we support async loading now, I'm not 100% convinced that we should warn in this condition. It seems like there could be cases where you accidentally or deliberately expect an icon to just work any time its related iconset loads. Maybe this type of warning should be opt-in (or opt-out) via an attribute? WDYT?